packages/shared-react/plugins/NodesMenu/index.tsx

```diff
@@ -0,0 +1,75 @@
+import { $getSelection, $isRangeSelection } from "lexical";
+import { useLexicalComposerContext } from "@lexical/react/LexicalComposerContext";
+import { useCallback, useEffect, useState } from "react";
+import FloatingBoxAtCursor from "../FloatingBox/FloatingBoxAtCursor";
+import { NodeOption, NodeSelectionMenu } from "./NodeSelectionMenu";
+
+export default function NodesMenu({ trigger, items }: { trigger: string; items?: NodeOption[] }) {
+  const [editor] = useLexicalComposerContext();
+  const [isOpen, setIsOpen] = useState(false);
+
+  // Close the menu when the escape key is pressed
+  const hideMenu = useCallback(
+    (e: KeyboardEvent) => {
+      if (e.key === "Escape") {
+        setIsOpen(false);
+        editor.focus();
+      }
+    },
+    [editor],
+  );
+
+  useEffect(() => {
+    if (isOpen) {
+      document.addEventListener("keydown", hideMenu);
+      return () => {
+        document.removeEventListener("keydown", hideMenu);
+      };
+    }
+  }, [isOpen, hideMenu]);
+
+  // Open the menu when the trigger key is pressed
+  useEffect(() => {
+    const showMenu = (e: KeyboardEvent) => {
```

Comment:
BTW I'm curious why this `showMenu` function is defined inside the `useEffect` whereas `hideMenu` is defined outside its `useEffect`?

Reply:
Probably because showMenu is only used inside this useEffect whereas hideMenu is used in two separate useEffect calls.